core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -171,5 +172,9 @@ object Consumer {
     }
   }
 
+  def committablePartitionedSource[K, V](settings: ConsumerSettings[K, V]):
+    Source[(TopicPartition, Source[CommittableMessage[K, V], Control]), Control] = {
```

Comment:
+1 to this source-of-sources solution. Having `control` should be enough for resource management. A new `Source` appearing in the stream is a much nice way to notify about reassignment. If we use `SubFlow` we can emit pairs of `(TopicPartition, CommittableMessage[K, V])`. Detecting partition assignment could be then possible by manual aggregation and analysis of incoming `TopicPartition` values which doesn't feel good.